app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/TransferSendManager.kt

```diff
@@ -78,72 +76,51 @@ class TransferSendManager @Inject constructor(
-                    _sendStatus.update { SendStatus.Failure }
-                }
-            },
-            onRefused = { _sendStatus.update { SendStatus.Refused } },
-        )
+        runCatching {
+            val attestationToken = getAttestationToken()
+            uploadManager.initUploadSession(
+                attestationHeaderName = AppIntegrityManager.ATTESTATION_TOKEN_HEADER,
+                attestationToken = attestationToken,
+            )!! // TODO: Handle ContainerErrorsException here
+            uploadWorkerScheduler.scheduleWork(uploadSessionUuid)
+            _sendStatus.update {
+                val totalSize = importationFilesManager.importedFiles.value.sumOf { it.fileSize }
+                SendStatus.Success(totalSize)
+            }
+        }.onFailure { exception ->
+            if (exception !is NetworkException && exception !is KmpNetworkException) {
+                SentryLog.e(TAG, "Failed to start the upload", exception)
+            }
+            if (exception is IntegrityException) {
+                _sendStatus.update { SendStatus.Refused }
+            } else {
+                _sendStatus.update { SendStatus.Failure }
+            }
+        }
     }
 
     //region App Integrity
-    private suspend inline fun withIntegrityToken(onSuccess: (attestationToken: String) -> Unit, onRefused: () -> Unit = {}) {
-        var attestationToken: String? = null
-
-        coroutineScope {
-            appIntegrityManager.getChallenge(
-                onSuccess = { launch { attestationToken = requestAppIntegrityToken(appIntegrityManager) } },
-                onFailure = {},
-            )
-        }
-
-        attestationToken?.let(onSuccess) ?: onRefused()
+    /** Handle [IntegrityException] and all the different exceptions that can be thrown */
+    private suspend inline fun getAttestationToken(): String {
+        val challenge = appIntegrityManager.getChallenge()
+        return requestAppIntegrityToken(challenge)
     }
 
-    private suspend fun requestAppIntegrityToken(appIntegrityManager: AppIntegrityManager): String? {
-        var attestationToken: String? = null
-
-        coroutineScope {
-            appIntegrityManager.requestClassicIntegrityVerdictToken(
-                onSuccess = { token ->
-                    SentryLog.i(APP_INTEGRITY_MANAGER_TAG, "request for app integrity token successful")
-                    launch { attestationToken = getApiIntegrityVerdict(appIntegrityManager, token) }
-                },
-                onFailure = {},
-            )
-        }
-
-        return attestationToken
+    private suspend fun requestAppIntegrityToken(challenge: String): String {
+        val token = appIntegrityManager.requestClassicIntegrityVerdictToken(challenge)
+        SentryLog.i(APP_INTEGRITY_MANAGER_TAG, "request for app integrity token successful")
+        return getApiIntegrityVerdict(appIntegrityManager, token)
```

Comment:
We should better name the two different tokens

```suggestion
      val appIntegrityToken = appIntegrityManager.requestClassicIntegrityVerdictToken(challenge)
        SentryLog.i(APP_INTEGRITY_MANAGER_TAG, "request for app integrity token successful")

        return getApiIntegrityVerdict(appIntegrityToken)
```